### PR TITLE
docs: Update Lifecycle.md

### PR DESCRIPTION
**Pull request by cw1997**
remove non-existent URL

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by jsumners — CHANGES_REQUESTED** [review]
Please correct the links instead of removing them.

File: docs/Reference/Lifecycle.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ being submitted, the data flow performed is the following:
 
 Note: `reply sent` means that the JSON payload will be serialized by:
 
-- the [reply serialized](./Server.md#setreplyserializer) if set
-- or by the [serializer compiler](./Server.md#setserializercompiler) when a JSON
+- the reply serialized if set
```
Comment (Uzlopak) [inline docs/Reference/Lifecycle.md:86]:
```suggestion
- the [reply serialized](./Server.md#set-reply-serializer) if set
```

File: docs/Reference/Lifecycle.md [hunk 2]
```diff
@@ -83,7 +83,7 @@ being submitted, the data flow performed is the following:
...
+- or by the serializer compiler when a JSON
```
Comment (Uzlopak) [inline docs/Reference/Lifecycle.md:87]:
```suggestion
- or by the [serializer compiler](./Server.md#set-serializer-compiler) when a JSON
```

**Comment by Uzlopak** [comment]
Actually the links are correct.

**Pull request closed without merging**